src/main/java/com/moddy/server/common/validation/prefer_regions/PreferRegionsValidator.java

```diff
@@ -0,0 +1,38 @@
+package com.moddy.server.common.validation.prefer_regions;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import java.util.List;
+
+public class PreferRegionsValidator implements ConstraintValidator<ValidPreferRegions, List<Long>> {
+
+    @Override
+    public void initialize(ValidPreferRegions constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(List<Long> preferRegions, ConstraintValidatorContext context) {
+
+        int zeroCount = 0;
+        int oneToFifteenCount = 0;
+
+        for (Long region : preferRegions) {
+            if (region == 0) {
+                zeroCount++;
+                if (zeroCount >= 1) {
+                    return false;
+                }
+            } else if (region >= 1 && region <= 25) {
+                oneToFifteenCount++;
+                if (oneToFifteenCount > 3) {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
p2

```suggestion
    if(preferRegions.size() > 3) return false;
    if(preferRegions.stream().filter(p -> p == 0).count() > 1)) return false;
    if(!preferRegions.stream().allMatch(p -> p >= 0 && p <= 26)) return false;
    return true;
```
p3
비록 순회를 2번 하여 2n 이지만, 가독성을 높인다는 측면에서 좋을 것 같다고 생각합니다!

Reply:
넵 그렇네요! 간단하게 바꿨습니다